### PR TITLE
Implement PRFM (register variant) as NOP

### LINKED ISSUE #1744: Missing PFRM (register) instruction
Requires the *PFRM (register)* instruction.</br>
```
The instruction at 0x0000000008093ca0 (opcode 0xf8a06a60) is undefined!
```
### Example dissasembly
```
PRFM            #0, [X3,X2]
```

### Required by: 
Only observed in homebrew

### LINKED ISSUE #1824: Missing prfm cpu instruction
### What CPU instruction is missing?

Requires the *prfm* instruction.</br>

```
The instruction at 0x00000000085fcd48 (opcode 0xf8a16801) is undefined!
```

### Instruction name
```
prfm pldl1strm, [x0, x1]
```

### Required by: 

https://github.com/Ryujinx/Ryujinx-Games-List/issues/401
https://github.com/Ryujinx/Ryujinx-Games-List/issues/3186

### PR DESCRIPTION
**Pull request by mageven**
Implements PRFM (register) as NOP, similar to other variants. Also, `pfrm -> prfm`

Closes #1744
Closes #1824 (duplicate)

Please test the programs/games linked in the issues as I don't have them.

### CODE REVIEW

**Comment by SeraUQ** [comment]
Confirmed `Edna & Harvey: Harvey's New Eyes` boots now and goes ingame, but the audio is messed up

**Comment by mageven** [comment]
Are you using OpenAL? It could be one of those games that sounds better with SoundIO (assuming you can use it)

**Comment by SeraUQ** [comment]
yeah tested with OpenAL, as in my setup SoundIO doesnt give any sound in any game I have tested with it

**Review by riperiperi — APPROVED** [review]
Nice, LGTM.

**Comment by EmulationFanatic** [comment]
I can't go in-game on Edna & Harvey with it in Windows. The intro and title screens sound fine on both backends, but it's very unstable. I got a crash during the intro nvdec scene followed by a reproducible system access violation exception crash at the title screen on the next attempt. I am uncertain if this is due to a bug in the implementation. Since @SeraUQ was able to go in-game in Linux, perhaps it is just me.

**Comment by EmulationFanatic** [comment]
Edna & Harvey: Harvey's New Eyes crashed on the intro on first launch, but seems like a one-time deal and/or PPTC fixes it. Now goes in-game with SoundIO working & sounds fine.
image

File: ARMeilleure/Decoders/OpCodeTable.cs [hunk 1]
```diff
@@ -144,9 +144,10 @@ static OpCodeTable()
...
             SetA64("10101010xx0xxxxxxxxxxxxxxxxxxxxx", InstName.Orr,             InstEmit.Orr,             OpCodeAluRs.Create);
-            SetA64("1111100110xxxxxxxxxxxxxxxxxxxxxx", InstName.Pfrm,            InstEmit.Pfrm,            OpCodeMemImm.Create);
-            SetA64("11111000100xxxxxxxxx00xxxxxxxxxx", InstName.Pfrm,            InstEmit.Pfrm,            OpCodeMemImm.Create);
-            SetA64("11011000xxxxxxxxxxxxxxxxxxxxxxxx", InstName.Pfrm,            InstEmit.Pfrm,            OpCodeMemLit.Create);
+            SetA64("1111100110xxxxxxxxxxxxxxxxxxxxxx", InstName.Prfm,            InstEmit.Prfm,            OpCodeMemImm.Create); // immediate
+            SetA64("11111000100xxxxxxxxx00xxxxxxxxxx", InstName.Prfm,            InstEmit.Prfm,            OpCodeMemImm.Create); // prfum (unscaled offset)
+            SetA64("11011000xxxxxxxxxxxxxxxxxxxxxxxx", InstName.Prfm,            InstEmit.Prfm,            OpCodeMemLit.Create); // literal
+            SetA64("11111000101xxxxxxxxx10xxxxxxxxxx", InstName.Prfm,            InstEmit.Prfm,            OpCodeMemReg.Create); // register
```
Comment (jduncanator) [inline ARMeilleure/Decoders/OpCodeTable.cs:150]: None of the other instructions have these comments, so they look a little out of place.

**Review by jduncanator — APPROVED** [review]
Looks good to me, maybe a small change.

**Comment by jduncanator** [comment]
This will also need a rebase for PPTC changes now.

File: ARMeilleure/Decoders/OpCodeTable.cs (hunk 1 above)
Comment (mageven) [reply ARMeilleure/Decoders/OpCodeTable.cs:150]: There are a few NEON instructions that have comments. Do I have to remove these? They kinda help in quickly figuring out the variant (not that it matters for this instruction)

**Comment by Shark-911** [comment]
Edna & Harvey: Harvey's New Eyes still crashes at the very beginning of the intro for me, PPTC is disabled though as it doesnt seem to work heres my log if theres anything useful in there for you.

[Ryujinx_1.0.6407_2021-01-25_18-12-32.log](https://github.com/Ryujinx/Ryujinx/files/5868694/Ryujinx_1.0.6407_2021-01-25_18-12-32.log)

**Comment by mageven** [comment]
> 
> 
> Edna & Harvey: Harvey's New Eyes still crashes at the very beginning of the intro for me, PPTC is disabled though as it doesnt seem to work heres my log if theres anything useful in there for you.
> 
> [Ryujinx_1.0.6407_2021-01-25_18-12-32.log](https://github.com/Ryujinx/Ryujinx/files/5868694/Ryujinx_1.0.6407_2021-01-25_18-12-32.log)

6407 is master. It won't have the instruction until this PR is merged. If you want to test it, you need to use one of the builds from this branch.

**Comment by EmulationFanatic** [comment]
Deponia is also booting now. Like Edna & Harvey, OpenAL sounds bad but SoundIO is fine. No crashes.
image

image

**Review by Thog — APPROVED** [review]

**Pull request merged**